### PR TITLE
config based cache and mem-cache-local module

### PR DESCRIPTION
**Pull request by shimonchayim**

### CODE REVIEW

File: modules/engine/test/cache-response-test.js [hunk 1]
```diff
@@ -402,7 +402,53 @@ module.exports = {
...
+            res.end(JSON.stringify({'counter':counter}));
+            counter++;
+        });
+        server.listen(3000, function () {
+            // Do the test here.
+            var engine = new Engine({
+                tables:__dirname + '/cache',
+                config:__dirname + '/cache/dev.json'
+            });
+            var script = "select * from auto.compute.key";
```
Comment (shimonchayim) [inline modules/engine/test/cache-response-test.js:428]: Plan to make a change which will require "expires" to be provided, next commit

**Comment by s3u** [comment]
Do we really need to add memcache binaries for testing? It will make tests less resilient on various op systems. 

Why not just a stub for testing calls?

**Comment by shimonchayim** [comment]
Memcache is only added for mem-cache-local testing for engine we have stubbed it out. If you check test/cache/dev.json you will see that we are using MockCache.

Regarding mem-cache-local. We need to discuss that.

**Comment by shimonchayim** [comment]
Status on my fork: Travis status

**Comment by shimonchayim** [comment]
Another comment, we are not adding memache binaries. I am including the source and building it.

**Pull request merged**